core/templates/dev/head/services/UserServiceSpec.ts

```diff
@@ -17,18 +17,79 @@
  */
 
 describe('User Service', function() {
-  var UserService, $httpBackend, UrlInterpolationService;
-
+  var UserService, $httpBackend, UrlInterpolationService,
+    UserInfoObjectFactory;
   beforeEach(angular.mock.module('oppia'));
   beforeEach(angular.mock.inject(function($injector) {
     UserService = $injector.get('UserService');
     UrlInterpolationService = $injector.get(
       'UrlInterpolationService');
+    UserInfoObjectFactory = $injector.get(
+      'UserInfoObjectFactory');
     $httpBackend = $injector.get('$httpBackend');
   }));
 
+
+  it('should be return userInfo data', function() {
+    // creating a test user for checking profile picture of user.
+    var sampleUserInfoBackendObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      user_is_logged_in: true
+    };
+
+    $httpBackend.expect('GET', '/userinfohandler').respond(200,
+      sampleUserInfoBackendObject);
+
+    var sampleUserInfo = UserInfoObjectFactory.createFromBackendDict(
+      sampleUserInfoBackendObject);
+
+    UserService.getUserInfoAsync().then(function(userInfo) {
+      expect(userInfo.is_admin).toBe(sampleUserInfo.is_admin);
+      expect(userInfo.is_super_admin).toBe(sampleUserInfo.is_super_admin);
+      expect(userInfo.is_moderator).toBe(sampleUserInfo.is_moderator);
+      expect(userInfo.is_topic_manager).toBe(sampleUserInfo.is_topic_manager);
+      expect(userInfo.user_is_logged_in).toBe(
+        sampleUserInfo.user_is_logged_in);
+      expect(userInfo.can_create_collections).toBe(
+        sampleUserInfo.can_create_collections);
+      expect(userInfo.username).toBe(sampleUserInfo.username);
+      expect(userInfo.preferred_site_language_code).toBe(
+        sampleUserInfo.preferred_site_language_code);
+    });
+    $httpBackend.flush();
+
+    // second call to /userinfohandler.
+    $httpBackend.expect('GET', '/userinfohandler').respond(200);
+    try {
+      // throw exception because /userinfohandler was not called
+      $httpBackend.flush();
+    } catch (err) {
+      expect(err.message).toBe('No pending request to flush !');
+    }
+  });
+
   it('should return image data', function() {
     var requestUrl = '/preferenceshandler/profile_picture';
+    // creating a test user for checking profile picture of user.
+    var sampleUserInfoBackendObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      user_is_logged_in: true
+    };
+    $httpBackend.expect('GET', '/userinfohandler').respond(200,
```

Comment:
Ditto here and below (re breaking after '(').

Reply:
Sorry I will take care now onwards. Done!